app/services/get_user_coordinates_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class GetUserCoordinatesService < BaseService
+  def initialize(params:)
+    super()
+    @ward = params[:ward]
+    @district = params[:district]
+  end
+
+  def call
+    validate!
+    return self if error?
+
+    fetch_request
+  end
+
+  private
+
+  def validate!
+    return add_error(I18n.t('errors.params.empty', param: :ward)) if @ward.blank?
```

Comment:
this is not correct since user can get their direction by either ward or district, the condition is not reasonable here